library/src/main/java/com/github/pwittchen/reactivewifi/WifiState.java

```diff
@@ -0,0 +1,52 @@
+package com.github.pwittchen.reactivewifi;
+
+/**
+ * Created by bhavdip on 2/9/17.
+ */
+
+public enum WifiState {
+  WIFI_STATE_DISABLING(0, "disabling"),
+  WIFI_STATE_DISABLED(1, "disabled"),
+  WIFI_STATE_ENABLING(2, "enabling"),
+  WIFI_STATE_ENABLED(3, "enabled"),
+  WIFI_STATE_UNKNOWN(4, "unknown");
+
+  public final int state;
+  public final String description;
+
+  WifiState(final int state, String description) {
+    this.state = state;
+    this.description = description;
+  }
+
+  /**
+   * Gets WifiState enum basing on integer value
+   *
+   * @param state as an integer
+   * @return WifiState enum
+   */
+  public static WifiState fromState(final int state) {
+    switch (state) {
+      case 0:
+        return WIFI_STATE_DISABLING;
+      case 1:
+        return WIFI_STATE_DISABLED;
+      case 2:
+        return WIFI_STATE_ENABLING;
+      case 3:
+        return WIFI_STATE_ENABLED;
+      case 4:
+        return WIFI_STATE_UNKNOWN;
+      default:
+        return WIFI_STATE_UNKNOWN;
+    }
+  }
+
+  @Override
+  public String toString() {
+    return "WifiState{" +
+        "state=" + state +
+        ", description='" + description + '\'' +
+        '}';
+  }
```

Comment:
`hashcode()` and `equals(...)` would be helpful in the case when we would like to filter reactive stream or compare different instances of the `WifiState`. Without the custom implementation of the hashcode and `equals(...)` it may not work properly. When you're using IntelliJ IDEA or Android Studio, you can generate these methods automatically by clicking `⌘N` on Mac or `alt+insert` on Win/Linux keyboard shortcut in the class. You can read more here: https://www.jetbrains.com/help/idea/2016.3/generate-equals-and-hashcode-wizard.html